lib/util/fs/index.ts

```diff
@@ -11,6 +11,18 @@ export * from './proxies';
 
 export const pipeline = util.promisify(stream.pipeline);
 
+function isPathInBaseDir(path: string, baseDir?: string): boolean {
+  if (baseDir && !path.startsWith(upath.resolve(baseDir))) {
```

Comment:
as base dir is always set at runtime we should make it mandatory to this function and instead use non null assertion on caller functions

Reply:
Alright, added this now in https://github.com/renovatebot/renovate/pull/16125/commits/f861c5513966236e49de0ac894a06dd07161c15f. Hope this is what you meant...